core/mm/mobj.c

```diff
@@ -141,15 +141,26 @@ static struct mobj *mobj_phys_init(paddr_t pa, size_t size, uint32_t cattr,
 {
 	void *va = NULL;
 	struct mobj_phys *moph = NULL;
+	struct tee_mmap_region *map = NULL;
 
 	if ((pa & CORE_MMU_USER_PARAM_MASK) ||
 	    (size & CORE_MMU_USER_PARAM_MASK)) {
 		DMSG("Expect %#x alignment", CORE_MMU_USER_PARAM_SIZE);
 		return NULL;
 	}
 
+	if (pa) {
+		va = phys_to_virt(pa, area_type);
+	} else {
+		map = core_mmu_find_mapping(area_type, size);
```

Comment:
What we need here is a function that can guarantee that we get the correct map. One way of doing that might be to require that there is only one `MEM_AREA_TEE_RAM_RW_DATA`.

Reply:
Sounds reasonable. I transformed core_mmu_find_mapping() to core_mmu_find_mapping_exclusive() to guarantee that found mapping is unique of that type.